uswds/uswds
# USWDS - Remove unused extract-loader dependency

# Summary

This PR removes the unused `extract-loader` dependency that was causing vulnerability through prototype pollution

## Breaking change

This is not a breaking change.

## Related issue

Resolves #5178

This PR should resolve https://github.com/uswds/uswds/security/dependabot/52

## Preview link

[Preview link →](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/cm-bump-JSON5/)

## Problem statement

Dependabot flagged prototype pollution in `JSON5: 0.5.1`.

## Solution

The `JSON5` version that was causing the vulnerability was a dependency of an older `babel-core` package, which in turn was a dependency of `extract-loader`.

```
├─┬ extract-loader@5.1.0
│ ├─┬ babel-core@6.26.3
│ │ └── json5@0.5.1 
```

After further investigation, it did not seem like our webpack config was referencing or using `extract-loader` at any point. I uninstalled the package and completed the following without error:

- Rebuilt USWDS
- Rebuilt storybook
- Rebuilt HTML previews
- Started local USWDS
- Served `_site` via Live Server
- `npm link` this USWDS package to USWDS Site where I rebuilt and started the local preview. No visible discrepancies through pages or component previews 

## Testing and review

- Perform the following commands and confirm there are no errors:

```
npm ci
npm build:storybook
npm build:html
npm start
```
- Inspect local build for visual errors
- Serve `_site` via Live Server and inspect for errors
- Test on Site:
   - In terminal, run`npm link`
   - Switch to your USWDS Site repo
   - In terminal run `npm link @uswds/uswds` to replace it's USWDS package with your link
   - Rebuild and start local build and check for errors
   - Inspect built site for errors


## Dependency updates

| Dependency name              | Previous version | New version |
| ---------------------------- | :--------------: | :---------: |
| extract-loader |     5.1.0     |     --      |

- Removed `extract-loader`

## Notes

- [extract-loader npm page](https://www.npmjs.com/package/extract-loader)
- extract-loader was added with storybook in PR #4295 

Before opening this PR, make sure you’ve done whichever of these applies to you:
